Assets/Scripts/Network/ColyseusClient.cs

```diff
@@ -93,142 +91,123 @@ public async void ConnectToServer(string serverEndpoint, string username, string
         {
-            StringBuilder sb = new StringBuilder();
-            sb.AppendLine("Connected clients:");
-            foreach (string p in players.Keys)
+            if (players.ContainsKey(message.playerId))
             {
-                sb.AppendLine(p);
+                if (message.updateType == NetworkMessageType.DeleteAnnotation.ToString())
+                {
+                    networkController.HandleAnnotationDelete(getPlayer(message.playerId), message.metadata);
+                }
+                else
+                {
+                    networkController.HandleNetworkInteraction(getPlayer(message.playerId), message.updateType);
+                }
             }
-            return sb.ToString();
-        }
-        else
-        {
-            return "";
-        }
+        });
     }
 
-    public NetworkPlayer GetPlayerById(string username)
+    NetworkPlayer getPlayer(string playerId)
     {
-        if (players != null && players.ContainsKey(username))
+        if (players.ContainsKey(playerId))
         {
-            return players[username];
+            return players.Values.First(p => p.id == playerId);
         }
         else return null;
     }
-    public void SendNetworkMessage(string message)
-    {
-        if (room != null)
-        {
-            room.Send(message);
-        }
-        else
-        {
-            CCDebug.Log("Room is not connected!", LogLevel.Warning, LogMessageCategory.Networking);
-        }
-    }
 
-    void OnMessage(object msg)
+    void OnMessageReceived(UpdateMessage message)
     {
-        if (msg is UpdateMessage)
+        CCDebug.Log("Received Schema Message!");
+        CCDebug.Log(message);
+        if (players.ContainsKey(message.playerId))
         {
-            // update messages have a message type and player Id we can use to update from remote interactions
-            var m = (UpdateMessage) msg;
-            CCDebug.Log(m.updateType + " " + m.playerId);
-            NetworkPlayer networkPlayer = players.Values.First(p => p.id == m.playerId);
-            if (m.updateType == "deleteannotation")
+            NetworkPlayer networkPlayer = players.Values.First(p => p.id == message.playerId);
+            if (message.updateType == NetworkMessageType.DeleteAnnotation.ToString())
             {
-                networkController.HandleAnnotationDelete(networkPlayer, m.metadata);
+                networkController.HandleAnnotationDelete(networkPlayer, message.metadata);
             }
             else
             {
-                networkController.HandleNetworkInteraction(networkPlayer, m.updateType);
+                networkController.HandleNetworkInteraction(networkPlayer, message.updateType);
             }
         }
-        else
+    }
+    async void LeaveRoom()
+    {
+        CCDebug.Log("closing connection");
+        await room.Leave(true);
+        room = null;
+    }
+
+    public NetworkPlayer GetPlayerById(string username)

```

Comment:
I remember being puzzled by this in the past: `GetPlayerById()` and `getPlayer()` above seem really similar.  I almost wonder if this method should be renamed getPlayerByUsername?

Reply:
Well spotted! I had to switch things up a bit in the update, so the key in the players dictionary is now the playerId, and it turned out we had two methods doing essentially the same thing (one more efficiently than the other!). Have now combined into one method called GetPlayerById.